packages/cli/test/unit/services/orchestration.service.test.ts

```diff
@@ -15,6 +15,21 @@ import { ActiveWorkflowManager } from '@/ActiveWorkflowManager';
 import { mockInstance } from '../../shared/mocking';
 import type { WorkflowActivateMode } from 'n8n-workflow';
 
+jest.mock('ioredis', () => {
+	const Redis = require('ioredis-mock');
+	if (typeof Redis === 'object') {
+		// the first mock is an ioredis shim because ioredis-mock depends on it
+		// https://github.com/stipsan/ioredis-mock/blob/master/src/index.js#L101-L111
+		return {
+			Command: { _transformer: { argument: {}, reply: {} } },
+		};
+	}
+	// second mock for our code
+	return function (...args: any) {
+		return new Redis(args);
+	};
+});
```

Comment:
we have a couple of places where we mock the redis connection like this because of how the code used to be written. We don't need to do this anymore, since we can now create a mock of `RedisClientService`.

```typescript
const redisClientService = mockInstance(RedisClientService);
const mockRedisClient = mock<Redis>();
redisClientService.createClient.mockReturnValue(mockRedisClient);
mockRedisClient.set.mockResolvedValue('OK');
```